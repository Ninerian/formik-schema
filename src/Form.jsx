import React from 'react';
import { Formik } from 'formik';

import { fetchRenderer } from './registry.jsx';

export class Form extends React.Component {
  constructor(props) {
    super(props);
    this.renderer = fetchRenderer(this.props.rendererName);
  }
  render () {
    return (
      <Formik
        {...this.props}
        render={this.renderer(this.props.schema)}
      />
    );
  }
}

