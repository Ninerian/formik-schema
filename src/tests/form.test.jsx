import React from 'react';
import Form from '../Form.jsx';
import renderer from 'react-test-renderer';


const testSchema2FormSnapshot = (testTitle, schema) => test(testTitle, () => {
  const component = renderer.create(
    <Form
      schema={schema}
      onSubmit={() => {}}
    >
    </Form>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

testSchema2FormSnapshot('Example from https://getbootstrap.com/docs/4.0/components/forms/#horizontal-form',
  {
    fields: [
      {
        name: "email",
        title: "Email",
        type: "email",
      },
      {
        name: "password",
        title: "Mot de passe",
        type: "password",
      },
      {
        name: "accept",
        title: "J'accepte.",
        type: "checkbox",
        description: "Check me out",
      },
    ],
  }
);

testSchema2FormSnapshot('Form with a choice and a default',
  {
    fields: [
      {
        name: "dog_or_cat",
        title: "Votre préférence ?",
        type: "choices",
        options: [
          {
            value: "cat",
            title: "Chat",
          },
          {
            value: "dog",
            title: "Chien",
            selected: true,
          },
        ],
      },
    ],
  }
);
