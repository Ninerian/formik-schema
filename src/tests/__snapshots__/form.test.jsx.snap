// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example from https://getbootstrap.com/docs/4.0/components/forms/#horizontal-form 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-2 col-form-label"
      for="email"
    >
      Email
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-2 col-form-label"
      for="password"
    >
      Mot de passe
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        type="password"
      />
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-2 col-form-label"
      for="accept"
    >
      J'accepte.
    </label>
    <div
      className="col-sm-10"
    >
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            className="form-check-input"
            onChange={[Function]}
            type="checkbox"
          />
           
          Check me out
        </label>
      </div>
    </div>
  </div>
</form>
`;

exports[`Form with a choice and a default 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-2 col-form-label"
      for="dog_or_cat"
    >
      Votre préférence ?
    </label>
    <div
      className="col-sm-10"
    >
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            id="dog_or_cat_cat"
            name="dog_or_cat"
            type="radio"
            value="cat"
          />
          Chat
        </label>
      </div>
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={true}
            className="form-check-input"
            id="dog_or_cat_dog"
            name="dog_or_cat"
            type="radio"
            value="dog"
          />
          Chien
        </label>
      </div>
    </div>
  </div>
</form>
`;
